lib/search_logic.rb

```diff
@@ -0,0 +1,133 @@
+module SearchLogic
+    class << self
+        def get_places_from_city_string(city_name, place_type)
+            begin
+                city = get_city_from_name(city_name)
+                places = city.public_send(place_type).to_a #city.attractions.to_a
+                return places if places.count == 20
+
+                case place_type
+                when "attractions"
+                    category = "tourism.attraction"
+                    response = get_place_details(category, city.city_place_id)
+                    create_attractions(response, city)
+                when "hotels"
+                    category = "accommodation.hotel"
+                    response = get_place_details(category, city.city_place_id)
+                    create_hotels(response, city)
+                when "restaurants"
+                    category = "catering.restaurant"
+                    response = get_place_details(category, city.city_place_id)
+                    create_restaurants(response, city)
+                end
+            rescue
+                raise PlaceTypeInvalid.new("Error: The place type used is not supported at the moment")
```

Comment:
A few things:

1. You rarely want to use `rescue` without a specific class, and with where this is placed, any error in the entire `begin` block (including `get_place_details`) would cause `PlaceTypeInvalid` to be raised, even if there's no problem with the place type given. There are some cases where a broad `rescue` okay, such as batch processing where you don't want an error with one record to halt the entire process, but even in those cases you'd often log the error.
2. For a better error, please indicate exactly what place type is invalid.
3. `begin` is redundant in Ruby when it encloses the entirety of a method or block.
4. There's a `case`/`when` right above, so I suggest appending an `else` to it and raising the exception there.